test/unit/uk/gov/hmrc/thirdpartyapplication/services/CredentialServiceSpec.scala

```diff
@@ -156,33 +157,37 @@ class CredentialServiceSpec extends AsyncHmrcSpec with ApplicationStateUtil {
 
     "add the client secret and return it unmasked in the name" in new Setup {
       ApplicationRepoMock.Fetch.thenReturn(applicationData)
-      ApplicationRepoMock.Save.thenAnswer((a: ApplicationData) => successful(a))
+
       val newSecretValue: String = "secret3"
       val maskedSecretValue: String = s"••••••••••••••••••••••••••••••••ret3"
-      when(mockGenerateSecret.apply()).thenReturn(newSecretValue)
+      val hashedSecret = newSecretValue.bcrypt
+      val newClientSecret = ClientSecret(maskedSecretValue, newSecretValue, hashedSecret = Some(hashedSecret))
 
-      val result = await(underTest.addClientSecret(applicationId, secretRequest))
+      ClientSecretServiceMock.GenerateClientSecret.thenReturnWithSpecificSecret(newSecretValue)
+
+      val updatedClientSecrets: List[ClientSecret] = applicationData.tokens.production.clientSecrets :+ newClientSecret
+      val updatedEnvironmentToken: EnvironmentToken = applicationData.tokens.production.copy(clientSecrets = updatedClientSecrets)
+      val updatedApplicationTokens = applicationData.tokens.copy(production = updatedEnvironmentToken)
+      val applicationWithNewClientSecret = applicationData.copy(tokens = updatedApplicationTokens)
+
+      ApplicationRepoMock.AddClientSecret.thenReturn(eqTo(applicationId), *)(applicationWithNewClientSecret)
+
+      val result: ApplicationTokenResponse = await(underTest.addClientSecret(applicationId, secretRequest))
 
-      val savedApp = ApplicationRepoMock.Save.verifyCalled()
-      val updatedProductionSecrets = savedApp.tokens.production.clientSecrets
-      updatedProductionSecrets should have size environmentToken.clientSecrets.size + 1
-      val newSecret = (updatedProductionSecrets diff environmentToken.clientSecrets).head
       result.clientId shouldBe environmentToken.clientId
       result.accessToken shouldBe environmentToken.accessToken
       result.clientSecrets.dropRight(1) shouldBe tokenResponse.clientSecrets
-      result.clientSecrets.last.secret shouldBe updatedProductionSecrets.last.secret
-      result.clientSecrets.last.name shouldBe newSecretValue
-      updatedProductionSecrets.last.name shouldBe maskedSecretValue
+      result.clientSecrets.last.secret shouldBe newSecretValue
+      result.clientSecrets.last.name shouldBe maskedSecretValue
```

Comment:
the name of the last client secret (the one just added) used to be unmasked (newSecretValue). Now it's masked. So devhub will display it masked all the time in the new UI (it can't rely on the secret field as that's going away).